scipy/stats/tests/test_stats.py

```diff
@@ -3230,112 +3234,163 @@ def _assert_equal(self, actual, expect, *, shape=None, dtype=None):
+        if dtype=='complex128' and is_torch(xp):
+            pytest.skip()
+        dtype = getattr(xp, dtype)
+        x = xp.asarray(np.random.rand(5), dtype=dtype)
         y = stats.moment(x, order=order)
-        self._assert_equal(y, expect, dtype=dtype)
+        xp_assert_equal(y, xp.asarray(expect, dtype=dtype))
 
-        y = stats.moment(np.broadcast_to(x, (6, 5)), axis=0, order=order)
-        self._assert_equal(y, expect, shape=(5,), dtype=dtype)
+        y = stats.moment(xp.broadcast_to(x, (6, 5)), axis=0, order=order)
+        xp_assert_equal(y, xp.full((5,), expect, dtype=dtype))
 
-        y = stats.moment(np.broadcast_to(x, (1, 2, 3, 4, 5)), axis=2,
+        y = stats.moment(xp.broadcast_to(x, (1, 2, 3, 4, 5)), axis=2,
                          order=order)
-        self._assert_equal(y, expect, shape=(1, 2, 4, 5), dtype=dtype)
+        xp_assert_equal(y, xp.full((1, 2, 4, 5), expect, dtype=dtype))
 
-        y = stats.moment(np.broadcast_to(x, (1, 2, 3, 4, 5)), axis=None,
+        y = stats.moment(xp.broadcast_to(x, (1, 2, 3, 4, 5)), axis=None,
                          order=order)
-        self._assert_equal(y, expect, shape=(), dtype=dtype)
+        xp_assert_equal(y, xp.full((), expect, dtype=dtype))
 
-    def test_moment_propagate_nan(self):
+    @array_api_compatible
+    def test_moment_propagate_nan(self, xp):
         # Check that the shape of the result is the same for inputs
         # with and without nans, cf gh-5817
         a = np.arange(8).reshape(2, -1).astype(float)
+        a = xp.asarray(a)
         a[1, 0] = np.nan
         mm = stats.moment(a, 2, axis=1, nan_policy="propagate")
-        np.testing.assert_allclose(mm, [1.25, np.nan], atol=1e-15)
+        xp_assert_close(mm, xp.asarray([1.25, np.nan], dtype=xp.float64), atol=1e-15)
 
-    def test_moment_empty_order(self):
+    @array_api_compatible
+    def test_moment_empty_order(self, xp):
         # tests moment with empty `order` list
         with pytest.raises(ValueError, match=r"'order' must be a scalar or a"
                                              r" non-empty 1D list/array."):
-            stats.moment([1, 2, 3, 4], order=[])
+            stats.moment(xp.asarray([1, 2, 3, 4]), order=[])
 
-    def test_rename_moment_order(self):
+    @array_api_compatible
+    def test_rename_moment_order(self, xp):
         # Parameter 'order' was formerly known as 'moment'. The old name
         # has not been deprecated, so it must continue to work.
-        x = np.arange(10)
+        x = xp.arange(10)
         res = stats.moment(x, moment=3)
         ref = stats.moment(x, order=3)
-        np.testing.assert_equal(res, ref)
+        xp_assert_equal(res, ref)
+
+    def test_moment_accuracy(self):
+        # 'moment' must have a small enough error compared to the slower
+        #  but very accurate numpy.power() implementation.
+        tc_no_mean = self.testcase_moment_accuracy - \
+                     np.mean(self.testcase_moment_accuracy)
```

Comment:
nit: I would prefer this but feel free to ignore
```suggestion
        tc_no_mean = (self.testcase_moment_accuracy
                      - np.mean(self.testcase_moment_accuracy))
```

Reply:
Not sure how my IDE got away with this. I never use explicit line breaks or put the operator on the previous line.